usl_pipeline/cloud_functions/main.py

```diff
@@ -1027,6 +1028,74 @@ def _direction_to_cosine(degrees):
     return dataset
 
 
+def _compute_solar_time_components(dataset: xarray.Dataset) -> xarray.Dataset:
+    # Solar Time Computation for every Longitude of the City
+    # Solar Time Sine and Cosine Derivation from there
+    if all(var in dataset.keys() for var in ["XLONG_M", "XLAT_M", "Times"]):
```

Comment:
How should we handle the case when these features are missing? Should we report an error to the user?